prog.go

```diff
@@ -869,6 +869,20 @@ func (p *Program) run(opts *RunOptions) (uint32, time.Duration, error) {
 		ctxOut = make([]byte, binary.Size(opts.ContextOut))
 	}
 
+	sysCtxOut := ctxOut
+	if ctxOut != nil && p.Type() == Syscall {
```

Comment:
FWIW I'd prefer sticking as close to upstream semantics as possible. They might allow ctx out at some point after all.

Just to make sure I understand correctly: would it be enough to do roughly:

```
if p.Type() == Syscall {
   // unmarshal from ctxIn
}
```

aftert the syscall has returned?